module/segment/src/Domain/Query/SegmentProductsQueryInterface.php

```diff
@@ -22,4 +22,9 @@ public function getProducts(SegmentId $segmentId): array;
      * @return array
      */
     public function getProductsByType(SegmentId $segmentId, string $type): array;
+
+    /**
+     * @return array
+     */
+    public function getAllEditedProducts(SegmentId $segmentId, ?\DateTime $dateTime = null): array;
```

Comment:
This is not a segment responsibility this query should be in exporter-File module